src/BSeries.jl

```diff
@@ -1897,32 +1910,45 @@ function _is_energy_preserving(trees, coefficients)
               Rational{Int8}, Rational{Int16}, Rational{Int32}, Rational{Int64},
               Rational{Int128}})
         # These types support efficient computations in sparse matrices
-        _is_energy_preserving_sparse(trees, coefficients)
+        _is_energy_preserving_sparse(trees, coefficients, atol, rtol)
     else
-        _is_energy_preserving_dense(trees, coefficients)
+        _is_energy_preserving_dense(trees, coefficients, atol, rtol)
     end
 end
 
-function _is_energy_preserving_dense(trees, coefficients)
+function _is_energy_preserving_dense(trees, coefficients, atol, rtol)
     # For every tree, obtain the adjoint and check if it exists
     length_coeff = length(trees)
     # Provided that a tree can have several adjoints, for every tree `t`
     # we need to check if there is a linear combination of the coefficients
     # of its adjoints such that this linear combination is equal to the
     # coefficient of `t`.
     # For this purpose, we create a energy_preserving_basis to store all the
-    # information
+    # information. We also include the tolerance:
     energy_preserving_basis = empty(trees)
     for (t_index, t) in enumerate(trees)
         # We do not need to consider the empty tree
         isempty(t) && continue
-
         # Check whether the level sequence corresponds to a bushy tree.
         # If so, we can exit early since the coefficients of bushy trees
         # must be zero in the modified equation of energy-preserving methods.
         if length(t) > 1 && is_bushy(t)
-            if !iszero(coefficients[t_index])
-                return false
+            if atol == 0 && rtol == 0
+                if !isapprox(coefficients[t_index],0)
+                    return false
+                end
+            elseif atol == 0 && rtol != 0
+                if !isapprox(coefficients[t_index],0; rtol=rtol)
+                    return false
+                end
+            elseif atol != 0 && rtol == 0
+                if !isapprox(coefficients[t_index],0; atol=atol)
+                    return false
+                end
+            else
+                if !isapprox(coefficients[t_index],0; atol=atol, rtol=rtol)
+                    return false
+                end
             end
```

Comment:
This looks very complicated to me - and contains quite a lot of branches that make it a bit difficult to check. Moreover, a relative tolerance alone does nothing when comparing something with zero (see above).